### PR TITLE
Proxify: make duplicate check optional

### PR DESCRIPTION
**Pull request by madsbk**
In order to improve performance, it is now possible to skip the duplication check in `ProxyManager.proxify()`.

We use this in explicit-comms shuffle.

### CODE REVIEW

File: dask_cuda/proxify_host_file.py [hunk 1]
```diff
@@ -322,20 +322,44 @@ def validate(self):
                         header, _ = pxy.obj
                         assert header["serializer"] == pxy.serializer
 
-    def proxify(self, obj: T) -> Tuple[T, bool]:
+    def proxify(self, obj: T, duplicate_check=True) -> Tuple[T, bool]:
         """Proxify `obj` and add found proxies to the `Proxies` collections
 
+        Search through `obj` and wraps all CUDA device objects in ProxyObject.
```
Comment (pentschev) [inline dask_cuda/proxify_host_file.py:328]:
```suggestion
        Search through `obj` and wrap all CUDA device objects in ProxyObject.
```

File: dask_cuda/proxify_host_file.py [hunk 2]
```diff
@@ -322,20 +322,44 @@ def validate(self):
...
         more incompatible-types were found.
+
+        Parameters
+        ----------
+        obj
+            Object to search through or wrap in a ProxyObject.
+        duplicate_check
+            Make sure that identical CUDA device objects found in `obj` are
+            wrapped by the same ProxyObject. This check comes with a significate
+            overhead hence set to False when it is known that no duplicate exist.
```
Comment (pentschev) [inline dask_cuda/proxify_host_file.py:342]:
```suggestion
            wrapped by the same ProxyObject. This check comes with a significant
            overhead hence it is recommended setting to False when it is known
            that no duplicate exist.
```

**Review by pentschev — APPROVED** [review]
LGTM @madsbk , left a few minor fixes/suggestions, otherwise good to go.

**Comment by madsbk** [comment]
Thanks @pentschev

**Pull request merged**